src/payment/strategies/paypal-commerce/paypal-commerce-payment-processor.spec.ts

```diff
@@ -0,0 +1,57 @@
+import { MissingDataError, MissingDataErrorType } from '../../../common/error/errors';
+
+import { OverlayPaypal, PaypalCommercePaymentProcessor } from './index';
+
+// tslint:disable:no-non-null-assertion
+describe('PaypalCommercePaymentProcessor', () => {
+    let paypalCommercePaymentProcessor: PaypalCommercePaymentProcessor;
+    let overlay: OverlayPaypal;
+    const focus = jest.fn();
+    const close = jest.fn();
+
+    beforeEach(() => {
+        window.open = jest.fn();
+
+        overlay = new OverlayPaypal();
+        paypalCommercePaymentProcessor = new PaypalCommercePaymentProcessor(overlay);
+    });
+
+    it('call window.open', async () => {
+        try {
+            await paypalCommercePaymentProcessor.paymentPayPal('approveUrl');
+        } catch (error) {
+            expect(window.open).toHaveBeenCalled();
+        }
+    });
+
+    it('throw error when window closed', async () => {
+        try {
+            await paypalCommercePaymentProcessor.paymentPayPal('approveUrl');
```

Comment:
for testing assertions, please prefer 
```js
await expect(paypalCommercePaymentProcessor.paymentPayPal('approveUrl')).rejects.toThrow(MissingDataError)
```
or
```js
expect(async () => await paypalCommercePaymentProcessor.paymentPayPal('approveUrl'))
  .toThrow(MissingDataError);
```

using try/catch should be avoided

Reply:
Thank you!